File: 墨奇0821_1.cpp
#include <algorithm>
#include <cmath>
#include <cstdio>
#include <cstring>
#include <iostream>
#include <vector>
using namespace std;

int n;
char s[200050];

void solve(int x) {
    vector<int> ans;
    ans.clear();

    int len = strlen(s);
    for (int i = 0; i < len; i++) {
        if (i == 0) {
            ans.push_back(1);
        }
        else {
            if (s[i] > s[i - 1]) {
                ans.push_back(ans[ans.size() - 1] + 1);
            }
            else {
                ans.push_back(1);
            }
        }
    }

    printf("Case #%d:", x);
    for (int i = 0; i < ans.size(); i++) {
        printf(" %d", ans[i]);
    }
    cout << endl;
}

int main() {
    int t;
    cin >> t;
    for (int i = 1; i <= t; i++) {
        cin >> n;
        cin >> s;
        solve(i);
    }
}

File: 墨奇0821_1.go
package main

import "fmt"

func solve(s string) []int {
	result := []int{}
	for i := 0; i < len(s); i++ {
		if i == 0 {
			result = append(result, 1)
		} else {
			if s[i] > s[i-1] {
				result = append(result, result[i-1]+1)
			} else {
				result = append(result, 1)
			}
		}
	}

	return result
}

func main() {
	var t int
	fmt.Scan(&t)

	var n int
	var s string
	ans := [][]int{}
	for t > 0 {
		t--
		fmt.Scan(&n)
		fmt.Scan(&s)
		ans = append(ans, solve(s))
	}

	for i := range ans {
		fmt.Printf("Case #%d:", i+1)
		for j := range ans[i] {
			fmt.Printf(" %d", ans[i][j])
		}
		fmt.Println()
	}
}
